### ISSUE TITLE
[REQUEST] Continuous session autosave

### ISSUE THREAD

**Issue by exp2ipi** [hlissner/doom-emacs#3487]
**Describe the feature**
Currently, Doom does not gracefully recover from unclean shutdowns. After a system crash, I discovered that `persp.el` workspace autosaves, as well as `recentf` file were both last written to over a week ago.  This is disappointing, as just about every other piece of modern tooling provides an option or a plugin to do regular checkpoint saves.  It's quite unfortunate that I can instantly restore my browser tabs and my tmux windows after a crash, but not my emacs state.

**Comment by Ruin0x11** [comment]
I tried adding this piece of code to my config:

```lua
(run-at-time (current-time) 300 'recentf-save-list)
```

But this only resulted in recentf being completely cleared after I restarted the server, even after a clean shutdown. So there's probably something else that doom-emacs does that hooks into recentf that makes it non-obvious what to do in order to autosave.

**Comment by exp2ipi** [comment]
> But this only resulted in recentf being completely cleared after I restarted the server...

Yeah, I noticed that too.  Life is ironic like that.  Here was my workaround:

```
(after! recentf
  (recentf-load-list)
  (run-at-time nil (* 120 60) #'recentf-save-list)) ; every 120 mins
```
I kind of wish the "wrote recentf" message didn't appear in the minibuffer, but other than that it works well enough.

**Comment by NightMachinery** [comment]
@betaprior What happens when two different emacs instances are open? Does `  (run-at-time nil (* 120 60) #'recentf-save-list))` make them overwrite each other?

**Comment by pivaldi** [comment]
I wrote [this Doom Emacs module](https://github.com/pivaldi/pimacs/tree/master/session) that provides rotating backup-ed sessions every hours and handles multiple simultaneous Doom Emacs instances with a lock file.

**Comment by hlissner** [comment]
This was resolved for recentf in 3b9aee686868d15ba02ad30e788550a3d81aea7f.

> What happens when two different emacs instances are open? Does   (run-at-time nil (* 120 60) #'recentf-save-list)) make them overwrite each other?

Naturally, they overwrite each other. `recentf` isn't designed to handle multiple instances.

-----

Anyhow, I won't be implementing continuous autosave for persp-mode, because it's much too brittle, but there's work going on right now to [replace persp-mode with tabspaces](https://github.com/doomemacs/modules/issues/15). Once that's done, I'll look into this further.

**Comment by hlissner** [comment]
Moving this to doomemacs/modules#15.
